### PR TITLE
Feature/gh 12358

### LINKED ISSUE #12358: Adapt top level group styling on overview page according to visuals.
The top level groups are currently styled the same as the child groups which irritates the user.
Style top level group according to visuals.

Collapsing of top level will be a separate task.

### PR DESCRIPTION
**Pull request by Changsuwan**
Change the styling of configuration overview area. Parent groups and children groups are now styled differently. 
Closes #12358

### CODE REVIEW

**Comment by Changsuwan** [comment]
To review this, select a multi-level product such as home-theater, configure and go to the overview tab.

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html [hunk 1]
```diff
@@ -47,3 +19,59 @@ <h2>{{ 'configurator.overviewForm.noAttributeHeader' | cxTranslate }}</h2>
...
+
+<ng-template #groups let-overviewGroups="overviewGroups" let-level="level">
+  <ng-container *ngFor="let group of overviewGroups">
+    <div
+      class="cx-group"
+      [class.topLevel]="level === 1"
+      [class.subgroupTopLevel]="level === 1 && group.subGroups?.length > 0"
+    >
+      <h2>
+        <ng-container *cxFeatureLevel="'!3.3'">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html]: As we want to release this with 4.0 or later, the cxFeatureLevel need to be removed

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-attribute.scss [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
   .cx-attribute-value {
     width: 100%;
+    font-weight: 700;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/styles/_configurator-overview-attribute.scss]: We need to document this change in  `/docs/migration/css-changes-in-version-4.md`

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss [hunk 3]
```diff
@@ -65,6 +82,9 @@
       }
 
       border: solid 1px var(--cx-color-light);
+      background-color: var(--cx-color-background);
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss:85]: Also those changes need to be documented

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html [hunk 4]
```diff
@@ -1,42 +1,14 @@
 <ng-container *ngIf="configuration$ | async as configuration">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html:1]: Is it really intended that sub groups and parent groups are styled that differently?
Screen Shot 2021-05-27 at 12 14 18

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-overview-normalizer.ts [hunk 5]
```diff
@@ -43,8 +43,7 @@ export class OccConfiguratorVariantOverviewNormalizer
     const characteristicValues: OccConfigurator.CharacteristicOverview[] =
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-overview-normalizer.ts:43]: Unit tests fail for OccConfiguratorVariantNormalizer

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html (hunk 4 above)
Comment (schuppell) [reply feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html:1]: Yes I will clarify it with Anthony today. But in the visuals they are styled differently.
In the future it is planned to collapse it on top level group. 
https://user-images.githubusercontent.com/51401884/117640208-10780580-b185-11eb-8492-1e58bccbd47e.png

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html (hunk 1 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html]: Done

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-attribute.scss (hunk 2 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/styles/_configurator-overview-attribute.scss]: Done

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss (hunk 3 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss:85]: Done

File: feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-overview-normalizer.ts (hunk 5 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-overview-normalizer.ts:43]: The tests have been fixed

File: docs/migration/css-changes-in-version-4.md [hunk 6]
```diff
@@ -5,3 +5,111 @@ title: Changes to Styles in 4.0
 ## Changes in Checkout Components
 
 * `cx-product-variants` selector has been moved to corresponding feature-lib `@spartacus/product`.
+
+## Change in Configurator Attribute Type Components
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md:9]: This has nothing to do with your activities, right? Why is it part of that PR?

Comment (ChristophHi) [reply docs/migration/css-changes-in-version-4.md:9]: The same for many other additions to this file! Please try to clean up a bit...

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: docs/migration/css-changes-in-version-4.md [hunk 7]
```diff
@@ -1,7 +1,32 @@
 ---
 title: Changes to Styles in 4.0
 ---
+## Changes in Configurator Overview Form Component
 
-## Changes in Checkout Components
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md:5]: Please do not remove this line

File: docs/migration/css-changes-in-version-4.md [hunk 8]
```diff
@@ -1,7 +1,32 @@
...
+* `font-size` set to 1.25rem on `%cx-configurator-overview-form` for `topLevel` selector to adjust the attribute header according to the new styling requirement 
 
-* `cx-product-variants` selector has been moved to corresponding feature-lib `@spartacus/product`.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md:7]: Please do not remove this line

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: docs/migration/css-changes-in-version-4.md (hunk 8 above)
Comment (Changsuwan) [reply docs/migration/css-changes-in-version-4.md:7]: The line has been added back

File: docs/migration/css-changes-in-version-4.md (hunk 7 above)
Comment (Changsuwan) [reply docs/migration/css-changes-in-version-4.md:5]: The line has been added back

**Review by ChristophHi — DISMISSED** [review]

**Review by ChristophHi — DISMISSED** [review]

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**